contrib/helm-charts/portus/templates/portus-configmap.yaml

```diff
@@ -99,70 +99,77 @@ data:
+        # identifier: ""
+        # # If a domain (e.g. mycompany.com) is set, then only signups with email from this domain are allowed.
+        # domain: ""
+{{ toYaml .open_id | indent 8 }}
 
       # Github authentication support.
       # Callback url: <host>/users/auth/github/callback
       github:
-        enabled: false
-        # Application credentials.
-        client_id: ""
-        client_secret: ""
-        # Only members of organization's team can sign in/up with Github.
-        organization: ""
-        team: ""
-        # If a domain (e.g. mycompany.com) is set, then only signups with email from this domain are allowed.
-        domain: ""
+        # enabled: false
+        # # Application credentials.
+        # client_id: ""
+        # client_secret: ""
+        # # Only members of organization's team can sign in/up with Github.
+        # organization: ""
+        # team: ""
+        # # If a domain (e.g. mycompany.com) is set, then only signups with email from this domain are allowed.
+        # domain: ""
+{{ toYaml .github | indent 8 }}
 
       # Gitlab authentication support.
       # Callback url: <host>/users/auth/gitlab/callback
       gitlab:
-        enabled: false
-        application_id: ""
-        secret: ""
-        # Only member of the group can sign in/up with Gitlab.
-        group: ""
-        # If a domain (e.g. mycompany.com) is set, then only signups with email from this domain are allowed.
-        domain: ""
-        # The Gitlab server to be used. If empty, then https://gitlab.com is assumed.
-        server: ""
+        # enabled: false
+        # application_id: ""
+        # secret: ""
+        # # Only member of the group can sign in/up with Gitlab.
+        # group: ""
+        # # If a domain (e.g. mycompany.com) is set, then only signups with email from this domain are allowed.
+        # domain: ""
+        # # The Gitlab server to be used. If empty, then https://gitlab.com is assumed.
+        # server: ""
+{{ toYaml .gitlab | indent 8 }}
 
       # Bitbucket authentication support. Need permission to read email.
       # Callback url: <host>/users/auth/bitbucket/callback
       bitbucket:
-        enabled: false
-        # Application credentials.
-        key: ""
-        secret: ""
-        # If a domain (e.g. mycompany.com) is set, then only signups with email from this domain are allowed.
-        domain: ""
-        options:
-          # Only members of team can sign in/up with Bitbucket. Need permission to read team membership.
-          team: ""
+        # enabled: false
+        # # Application credentials.
+        # key: ""
+        # secret: ""
+        # # If a domain (e.g. mycompany.com) is set, then only signups with email from this domain are allowed.
+        # domain: ""
+        # options:
+        #   # Only members of team can sign in/up with Bitbucket. Need permission to read team membership.
+        #   team: ""
+{{ toYaml .bitbucket | indent 8 }}
```

Comment:
Nice and compact, though as far as flexibility goes this could end up being less so because the values per section are just dumped to this file as is from values.yaml. With that said, I can't think of a good reason why it matters in this particular case. Example of if case: https://github.com/kubic-project/caasp-services/blob/master/contrib/helm-charts/portus/templates/portus-deployment.yaml#L43

@mssola any opinion here?

Reply:
For example, items in `google_oauth2` are nested.

> For optimal safety, a nested value must be checked at every level.

I agree with this.

https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/values.md#flat-or-nested-values

I just dumped the values because it made me feel uncomfortable to flatten the items in values.yaml.

But I can understand what you say.

Reply:
For example, items in `google_oauth2` are nested.

> For optimal safety, a nested value must be checked at every level.

I agree with this.

https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/values.md#flat-or-nested-values

I just dumped the values because it made me feel uncomfortable to flatten the items in values.yaml.

But I can understand what you say.